### DISCUSSION TITLE
请问有语法错误时候msfu热更新中断退出怎么办？

### DISCUSSION THREAD

**Discussion by fireworksflyaway** [umijs/umi#9601] [Q&A] [answered]
umi max官方脚手架+pnpm，随便在某个组件里面写个语法错误，然后等一两秒，热更新显示错误，然后就退出了，需要重新pnpm dev，以前用3.0时候也是这样，现在4.0还是这样，是我使用姿势不对吗？
以前一直用react的devServer，这种语法错误热更新不会中断啊

![捕获](https://user-images.githubusercontent.com/9814895/197338562-2884d9da-7597-48bb-aa9f-0eb7a2bf29ac.JPG)

**Comment by fz6m** [comment]
@stormslowly 可以看下这个问题，当某个源码 `src/*` 文件对应的 mfsu 缓存（ `.js` 文件 ）不存在时，并且 `.ts` 源码是语法错误的文件，项目无法正常启动。

比如创建一个 `src/app.ts` （需要保证之前没有创建过该文件） ：

```ts
eqwqwqwq
```

之后 `pnpm dev` 启动项目将崩溃。

**Reply by fz6m** [reply]
另外 `你知道吗？` 会被运行两次，有一次是在 worker 里么，这个也得去一次好像

**Reply by fz6m** [reply]
最新版本 `4.0.28` 以上问题均已解决。

**Reply by fireworksflyaway** [reply]
试了一下4.0.28，其他文件语法错误好像不会热更新中断退出，但是我这边src/app.tsx这个文件如果有语法错误还是会中断退出

**Reply by fz6m** [reply]
有空的时候 @stormslowly 可以看下这个 MFSU 问题，复现步骤：

1. 删除 `node_modules/.cache`

2. 添加 `src/app.ts` ，注意数字开头再加英文 ：
    
    ```ts
    // app.ts
    
    3222eeeeee
    ```

3. 启动服务：
    
    ```bash
      pnpm dev
    ```

**Reply by fz6m** [reply]
类似的问题见：https://github.com/umijs/umi/issues/9667

**Reply by coding-ice** [reply]
请问这个问题有解决方案吗～ 语法错误导致的终端退出

**Comment by stormslowly** [comment, marked as answer]
与 MFSU 无关

app.ts 作为项目的运行时配置文件 在项目启动时需要解析来决定生成文件；如果 app.ts 有语法错误，无法解析，关闭 dev server 符合预期；类似 .umirc.ts 有错误项目也起不来。


duplicated https://github.com/umijs/umi/issues/9678
